test/controllers/UpfrontPaymentControllerSpec.scala

```diff
@@ -16,27 +16,112 @@
     }
   }
+
+  "POST /can-you-make-an-upfront-payment [Yes] redirects to upfront-payment-amount" in {
+    AuthStub.authorise()
+    EssttpBackend.EligibilityCheck.findJourneyAfterEligibilityCheck
+    EssttpBackend.CanPayUpfront.updateCanPayUpfront(TdAll.journeyId, canPayUpfrontScenario = true)
+
+    val fakeRequest = FakeRequest(
+      method = "POST",
+      path   = "/can-you-make-an-upfront-payment"
+    ).withAuthToken()
+      .withSession(SessionKeys.sessionId -> "IamATestSessionId")
+      .withFormUrlEncodedBody(("CanYouMakeAnUpFrontPayment", "Yes"))
+
+    val result: Future[Result] = controller.canYouMakeAnUpfrontPaymentSubmit(fakeRequest)
+    status(result) shouldBe Status.SEE_OTHER
+    redirectLocation(result) shouldBe Some("/set-up-a-payment-plan/upfront-payment-amount")
+  }
+
+  "POST /can-you-make-an-upfront-payment [No] redirects to monthly payment amount" in {
+    AuthStub.authorise()
+    EssttpBackend.EligibilityCheck.findJourneyAfterEligibilityCheck
+    EssttpBackend.CanPayUpfront.updateCanPayUpfront(TdAll.journeyId, canPayUpfrontScenario = false)
+
+    val fakeRequest = FakeRequest(
+      method = "POST",
+      path   = "/can-you-make-an-upfront-payment"
+    ).withAuthToken()
+      .withSession(SessionKeys.sessionId -> "IamATestSessionId")
+      .withFormUrlEncodedBody(("CanYouMakeAnUpFrontPayment", "No"))
+
+    val result: Future[Result] = controller.canYouMakeAnUpfrontPaymentSubmit(fakeRequest)
+    status(result) shouldBe Status.SEE_OTHER
+    redirectLocation(result) shouldBe Some("/set-up-a-payment-plan/monthly-payment-amount")
+  }
+
+  "POST /can-you-make-an-upfront-payment [No option selected] returns /can-you-make-an-upfront-payment with error summary" in {
+    AuthStub.authorise()
+    EssttpBackend.EligibilityCheck.findJourneyAfterEligibilityCheck
+
+    val fakeRequest = FakeRequest(
+      method = "POST",
+      path   = "/can-you-make-an-upfront-payment"
+    ).withAuthToken().withSession(SessionKeys.sessionId -> "IamATestSessionId")
+
+    val result: Future[Result] = controller.canYouMakeAnUpfrontPaymentSubmit(fakeRequest)
+
+    status(result) shouldBe Status.OK
+    contentType(result) shouldBe Some("text/html")
+    charset(result) shouldBe Some("utf-8")
+
+    val pageContent: String = contentAsString(result)
+    val doc: Document = Jsoup.parse(pageContent)
+
+    doc.title() shouldBe s"Error $expectedPageTitle"
+    doc.select(".govuk-fieldset__heading").text() shouldBe expectedH1
+    doc.select(".hmrc-header__service-name").text() shouldBe expectedServiceName
+    doc.select(".hmrc-sign-out-nav__link").attr("href") shouldBe "http://localhost:9949/auth-login-stub/session/logout"
+    doc.select("#CanYouMakeAnUpFrontPayment-hint").text() shouldBe expectedPageHint
+    doc.select("#main-content > div > div > div > div > div > ul > li > a").text() shouldBe "Select yes if you can make an upfront payment"
```

Comment:
a nicer way to select this would be something like this:
```scala
 val errorSummary = doc.select(".govuk-error-summary")
 val errorLink = errorSummary.select("a")
 errorLink.text() shouldBe "Select blah..."
 errorLink.attr("href") shouldBe "#CanYouMakeBlah"
```